ui/nuxeo-document-comments/nuxeo-document-comment-thread.js

```diff
@@ -0,0 +1,293 @@
+import '@nuxeo/nuxeo-elements/nuxeo-resource.js';
+import '../widgets/nuxeo-tooltip.js';
+import '../widgets/nuxeo-user-avatar.js';
+// eslint-disable-next-line import/no-cycle
+import './nuxeo-document-comment.js';
+import './nuxeo-document-comments-styles.js';
+import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { FormatBehavior } from '../nuxeo-format-behavior.js';
+
+/**
+ `nuxeo-document-comment-thread`
+ @group Nuxeo UI
+ @element nuxeo-document-comment-thread
+ */
+{
+  class DocumentCommentThread extends mixinBehaviors([FormatBehavior], Nuxeo.Element) {
+    static get template() {
+      return html`
+        <style include="nuxeo-document-comments-styles"></style>
+
+        <nuxeo-resource id="commentRequest" path="/id/[[uid]]/@comment/"></nuxeo-resource>
+
+        <template is="dom-if" if="[[_moreAvailable(comments.length, total, allCommentsLoaded)]]">
+          <span class="more-content no-selection pointer smaller" on-tap="_loadMore"
+            >[[_computeTextLabel(level, 'loadAll', total, i18n)]]</span
+          >
+        </template>
+        <template id="commentList" is="dom-repeat" items="[[comments]]" as="comment">
+          <nuxeo-document-comment comment="{{comment}}" level="[[level]]"></nuxeo-document-comment>
+        </template>
+
+        <template is="dom-if" if="[[_allowReplies(level)]]">
+          <div class="input-area">
+            <paper-textarea
+              id="inputContainer"
+              placeholder="[[_computeTextLabel(level, 'writePlaceholder', null, i18n)]]"
+              value="{{text}}"
+              max-rows="[[_computeMaxRows()]]"
+              no-label-float
+              on-keydown="_checkForEnter"
+            >
+            </paper-textarea>
+            <template is="dom-if" if="[[!_isBlank(text)]]">
+              <iron-icon
+                id="submit"
+                name="submit"
+                class="main-option opaque"
+                icon="check"
+                on-tap="_submitComment"
+              ></iron-icon>
+              <nuxeo-tooltip for="submit">[[i18n('comments.submit.tooltip')]]</nuxeo-tooltip>
+              <iron-icon name="clear" class="main-option opaque" icon="clear" on-tap="_clearInput"></iron-icon>
+            </template>
+          </div>
+        </template>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-comment-thread';
+    }
+
+    static get properties() {
+      return {
+        uid: {
+          type: String,
+          observer: '_refresh',
+        },
+
+        comments: {
+          type: Array,
+          value() {
+            return [];
+          },
+        },
+
+        level: {
+          type: Number,
+          value: 1,
+        },
+
+        pageSize: {
+          type: Number,
+          readOnly: true,
+          value: 10,
+        },
+
+        allCommentsLoaded: {
```

Comment:
nit: reflecting this to attribute could be useful for styling, although we don't  really have a need for it right now...